integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonInsertFromStageCommand.scala

```diff
@@ -431,8 +318,53 @@ case class CarbonInsertFromStageCommand(
     }.map { future =>
       future.get()
     }
+  }
+
+  /**
+   * Delete stage file and success file with retry
+   */
+  private def deleteStageFilesWithRetry(
+      executorService: ExecutorService,
+      _stageFiles: Array[(CarbonFile, CarbonFile)]): Unit = {
+    val startTime = System.currentTimeMillis()
+    var retryCount = DELETE_STAGEFILES_RETRY_TIMES
+    var stageFiles = _stageFiles
+    while (stageFiles.length > 0 && retryCount > 0) {
+      retryCount -= 1
+      // 1) delete stage files
+      deleteStageFiles(executorService, stageFiles)
+      // 2) ensure stage files are all deleted
+      stageFiles = ensureStageFilesAreAllDeleted(executorService, stageFiles)
```

Comment:
I think this will slow down the performance, can we check it by using the return value of delete operation in `deleteStageFiles`?

Reply:
solved